include/appfwk/DAQModule.hpp

```diff
@@ -99,10 +156,34 @@ class DAQModule : public NamedObject
    *  Non-accepted commands or failure should return an ERS exception
    * indicating this result.
    */
-  virtual void execute_command(const std::string& cmd, const std::vector<std::string>& args) = 0;
+  void execute_command(const std::string& name, const std::vector<std::string>& args = {});
+
+  std::vector<std::string> get_commands() const;
+
+  bool has_command(const std::string name) const;
 
 protected:
+  /**
+   * @brief Registers a mdoule command under the name `cmd`.
+   * Returns whether the command was inserted (false meaning that command `cmd` already exists)
+   */
+  template <typename Child>
+  void
+  register_command(const std::string &name, void (Child::*f)(const std::vector<std::string>&)) {
```

Comment:
This is a complex function: let's leave the implementation in the hxx and let's not change the order of the file. class definition first, issues last. No need to change the whole file only to have an inline function.

Reply:
Fair enough, let me do that.
Anyway, we need to sort out why doxygen and ers don't get along. I'll open an issue ™️